src/API/ApiEditSlots.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace WSSlots\API;
+
+use ApiBase;
+use ApiUsageException;
+use MediaWiki\Storage\SlotRecord;
+use MWContentSerializationException;
+use MWException;
+use Wikimedia\ParamValidator\ParamValidator;
+use WSSlots\Logger;
+use WSSlots\WSSlots;
+use MediaWiki\MediaWikiServices;
+
+/**
+ * A slot-aware module that allows for editing and creating pages.
+ */
+class ApiEditSlots extends ApiBase {
+	/**
+	 * @inheritDoc
+	 *
+	 * @throws ApiUsageException
+	 * @throws MWContentSerializationException
+	 * @throws MWException
+	 */
+	public function execute() {
+		$this->useTransactionalTimeLimit();
+
+		$user = $this->getUser();
+		$params = $this->extractRequestParams();
+		$wikiPage = $this->getTitleOrPageId( $params );
+		$title = $wikiPage->getTitle();
+		$apiResult = $this->getResult();
+
+		// Check if we are allowed to edit or create this page
+		$this->checkTitleUserPermissions(
+			$title,
+			$title->exists() ? 'edit' : [ 'edit', 'create' ],
+			[ 'autoblock' => true ]
+		);
+
+		$slotUpdates = [];
+
+        $slots = MediaWikiServices::getInstance()->getSlotRoleRegistry()->getKnownRoles();
+        foreach ( $slots as $slotName ) {
+			if ( isset( $params[ self::maskSlotName( $slotName ) ] ) ) {
+				$slotUpdates[ $slotName ] = $params[ self::maskSlotName( $slotName ) ];
+			}
+		}
+
+		$result = WSSlots::editSlots(
+			$user,
+			$wikiPage,
+			$slotUpdates,
+			$params["summary"],
+			$params["append"],
+			$params["watchlist"]
+		);
+
+		if ( $result !== true ) {
+			list( $message, $code ) = $result;
+
+			Logger::getLogger()->alert( 'Editing slot failed while performing edit through the "editslots" API: {message}', [
+				'message' => $message
+			] );
+
+			$this->dieWithError( $message, $code );
+		} else {
+			$apiResult->addValue( null, 'editslots', ['result' => 'success'] );
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function mustBePosted(): bool {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function isWriteMode(): bool {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getAllowedParams(): array {
+		$params = [
+			'title' => [
+				ApiBase::PARAM_TYPE => 'string'
```

Comment:
deprecated

Reply:
I will leave this for another PR, since it would require changing the other API modules as well.